src/interactions/slashCommands/Assertions.ts

```diff
@@ -0,0 +1,34 @@
+import type { APIApplicationCommandOptionChoice } from 'discord-api-types/v8';
+import ow from 'ow';
+import type { ToAPIApplicationCommandOptions } from './SlashCommandBuilder';
+
+export function validateRequiredParameters(
+	name: string,
+	description: string,
+	options: ToAPIApplicationCommandOptions[],
+) {
+	// Assert name matches all conditions
+	validateName(name);
+
+	// Assert description conditions
+	validateDescription(description);
+
+	// Assert options conditions
+	validateMaxOptionsLength(options);
+}
+
+export function validateName(name: unknown): asserts name is string {
+	ow(name, 'name', ow.string.lowercase.alphabetical.minLength(1).maxLength(32));
```

Comment:
This validation incorrectly rejects, for example, `grüße` and `どうも`.
`alphabetical` is probably a bit too strict.

Maybe the [PR about the name regex](https://github.com/discord/discord-api-docs/pull/2901) in discord's docs can help you further here, since I fear I can't.